src/set_theory/game/pgame.lean

```diff
@@ -261,287 +249,284 @@ instance : has_one pgame := ⟨⟨punit, pempty, λ _, 0, pempty.elim⟩⟩
+by { rw le_def_lf, dsimp, simp }
 
-theorem lt_of_le_move_left {x y : pgame} {i} :
-  x ≤ y.move_left i → x < y :=
-by { cases y, rw move_left_mk, exact lt_mk_of_le }
+/-- The definition of `x ≤ 0` on pre-games, in terms of `⧏ 0`. -/
+theorem le_zero_lf {x : pgame} : x ≤ 0 ↔ ∀ i, x.move_left i ⧏ 0 :=
+by { rw le_def_lf, dsimp, simp }
 
-private theorem not_le_lt {x y : pgame} :
-  (¬ x ≤ y ↔ y < x) ∧ (¬ x < y ↔ y ≤ x) :=
-begin
-  induction x with xl xr xL xR IHxl IHxr generalizing y,
-  induction y with yl yr yL yR IHyl IHyr,
-  classical,
-  simp only [mk_le_mk, mk_lt_mk,
-    not_and_distrib, not_or_distrib, not_forall, not_exists,
-    and_comm, or_comm, IHxl, IHxr, IHyl, IHyr, iff_self, and_self]
-end
+/-- The definition of `0 ⧏ x` on pre-games, in terms of `0 ≤`. -/
+theorem zero_lf_le {x : pgame} : 0 ⧏ x ↔ ∃ i, 0 ≤ x.move_left i :=
+by { rw lf_def_le, dsimp, simp }
 
-@[simp] theorem not_le {x y : pgame} : ¬ x ≤ y ↔ y < x := not_le_lt.1
-@[simp] theorem not_lt {x y : pgame} : ¬ x < y ↔ y ≤ x := not_le_lt.2
+/-- The definition of `x ⧏ 0` on pre-games, in terms of `≤ 0`. -/
+theorem lf_zero_le {x : pgame} : x ⧏ 0 ↔ ∃ j, x.move_right j ≤ 0 :=
+by { rw lf_def_le, dsimp, simp }
 
-@[refl] protected theorem le_rfl : ∀ {x : pgame}, x ≤ x
-| ⟨l, r, L, R⟩ := by rw mk_le_mk; exact
-⟨λ i, lt_mk_of_le le_rfl, λ i, mk_lt_of_le le_rfl⟩
+/-- The definition of `0 ≤ x` on pre-games, in terms of `0 ≤` two moves later. -/
+theorem zero_le {x : pgame} : 0 ≤ x ↔ ∀ j, ∃ i, 0 ≤ (x.move_right j).move_left i :=
+by { rw le_def, dsimp, simp }
 
-protected theorem le_refl (x : pgame) : x ≤ x :=
-pgame.le_rfl
+/-- The definition of `x ≤ 0` on pre-games, in terms of `≤ 0` two moves later. -/
+theorem le_zero {x : pgame} : x ≤ 0 ↔ ∀ i, ∃ j, (x.move_left i).move_right j ≤ 0 :=
+by { rw le_def, dsimp, simp }
 
-protected theorem lt_irrefl (x : pgame) : ¬ x < x :=
-not_lt.2 pgame.le_rfl
+/-- The definition of `0 ⧏ x` on pre-games, in terms of `0 ⧏` two moves later. -/
+theorem zero_lf {x : pgame} : 0 ⧏ x ↔ ∃ i, ∀ j, 0 ⧏ (x.move_left i).move_right j :=
+by { rw lf_def, dsimp, simp }
 
-protected theorem ne_of_lt : ∀ {x y : pgame}, x < y → x ≠ y
-| x _ h rfl := pgame.lt_irrefl x h
+/-- The definition of `x ⧏ 0` on pre-games, in terms of `⧏ 0` two moves later. -/
+theorem lf_zero {x : pgame} : x ⧏ 0 ↔ ∃ j, ∀ i, (x.move_right j).move_left i ⧏ 0 :=
+by { rw lf_def, dsimp, simp }
 
-/-- In general, `xL i ≤ x` isn't true. It is true however for `numeric` games, see
-`numeric.move_left_le`. -/
-theorem lt_mk {xl xr : Type u} {xL : xl → pgame} {xR : xr → pgame} (i) : xL i < mk xl xr xL xR :=
-lt_mk_of_le pgame.le_rfl
+@[simp] theorem zero_le_of_is_empty_right_moves (x : pgame) [is_empty x.right_moves] : 0 ≤ x :=
+zero_le.2 is_empty_elim
 
-/-- In general, `x ≤ xR i` isn't true. It is true however for `numeric` games, see
-`numeric.move_right_le`. -/
-theorem mk_lt {xl xr : Type u} {xL : xl → pgame} {xR : xr → pgame} (i) : mk xl xr xL xR < xR i :=
-mk_lt_of_le pgame.le_rfl
```

Comment:
Seems these three are removed?

Reply:
I removed a bunch of `mk` statements for the order relations. This is because the non-`mk` versions are strictly more general.

Now that you point this out, I could've just left them be and remove them in a subsequent PR. Should I do this?

Reply:
I see that the removed versions are defeq to the versions that remain; I think if these are `=` of `iff` that can be used to rewrite, then the syntactic form matters and usually we'd keep both. However these are neither, so I'm not sure whether we should keep both (does `library_search` sees through the defeq, or the syntactic form matters)?

Reply:
`library_search` does not see through them. And now that I've tested them a bit, it seems like you sometimes need to give the explicit argument `x` for it to work. These seem like good enough reasons to have otherwise duplicate lemmas.